bundles/org.openhab.binding.echonetlite/src/main/java/org/openhab/binding/echonetlite/internal/EchonetLiteBridgeHandler.java

```diff
@@ -0,0 +1,401 @@
+        if (null == devicesByKey.putIfAbsent(device.instanceKey(), device)) {
+            logger.debug("New device discovered: {}", device.instanceKey);
+        }
+    }
+
+    private void pollDevices(long nowMs, EchonetChannel echonetChannel) {
+        for (EchonetObject echonetObject : devicesByKey.values()) {
+            if (echonetObject.buildUpdateMessage(messageBuilder, echonetChannel::nextTid, nowMs,
+                    requireNonNull(managementControllerKey))) {
+                try {
+                    echonetChannel.sendMessage(messageBuilder);
+                } catch (IOException e) {
+                    logger.warn("Failed to send echonet message", e);
+                }
+            }
+
+            echonetObject.refreshAll(nowMs);
+
+            if (echonetObject.buildPollMessage(messageBuilder, echonetChannel::nextTid, nowMs,
+                    requireNonNull(managementControllerKey))) {
+                try {
+                    echonetChannel.sendMessage(messageBuilder);
+                } catch (IOException e) {
+                    logger.warn("Failed to send echonet message", e);
+                }
+            }
+        }
+    }
+
+    private void pollRequests() {
+        Message message;
+        while (null != (message = requestsPoll())) {
+            logger.debug("Received request: {}", message);
+            if (message instanceof NewDeviceMessage) {
+                newDeviceInternal((NewDeviceMessage) message);
+            } else if (message instanceof RefreshMessage) {
+                refreshDeviceInternal((RefreshMessage) message);
+            } else if (message instanceof RemoveDevice) {
+                removeDeviceInternal((RemoveDevice) message);
+            } else if (message instanceof UpdateDevice) {
+                updateDeviceInternal((UpdateDevice) message);
+            } else if (message instanceof StartDiscoveryMessage) {
+                startDiscoveryInternal((StartDiscoveryMessage) message);
+            } else if (message instanceof StopDiscoveryMessage) {
+                stopDiscoveryInternal((StopDiscoveryMessage) message);
+            }
+        }
+    }
+
+    private @Nullable Message requestsPoll() {
+        return requests.poll();
+    }
+
+    private void pollNetwork(EchonetChannel echonetChannel) {
+        try {
+            echonetChannel.pollMessages(echonetMessage, this::onMessage,
+                    EchonetLiteBindingConstants.NETWORK_WAIT_TIMEOUT);
+        } catch (IOException e) {
+            logger.warn("Failed to poll for messages", e);
+        }
+    }
+
+    private void onMessage(final EchonetMessage echonetMessage) {
+        final EchonetClass echonetClass = echonetMessage.sourceClass();
+        final SocketAddress sourceAddress = echonetMessage.sourceAddress();
+        if (null == echonetClass) {
+            logger.warn("Unable to find echonetClass for message: {}", echonetMessage.toDebug());
```

Comment:
When users see this, should they report it as a bug, or what is needed to mitigate this? Should the bridge remain **ONLINE**?